src/osal/lv_cmsis_rtos2.c

```diff
@@ -0,0 +1,190 @@
+/**
+ * @file lv_cmsis_rtos2.c
+ *
+ */
+
+/*
+ * Copyright (C) 2023 Arm Limited or its affiliates. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "lv_os.h"
+
+#if LV_USE_OS == LV_OS_CMSIS_RTOS2
+
+#include "../misc/lv_log.h"
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+lv_res_t lv_thread_init(lv_thread_t * thread, lv_thread_prio_t prio, void (*callback)(void *), size_t stack_size,
+                        void * user_data)
+{
+    static const osPriority_t prio_map[] = {
+        [LV_THREAD_PRIO_LOWEST] = osPriorityLow,
+        [LV_THREAD_PRIO_LOW] = osPriorityBelowNormal,
+        [LV_THREAD_PRIO_MID] = osPriorityNormal,
+        [LV_THREAD_PRIO_HIGH] = osPriorityHigh,
+        [LV_THREAD_PRIO_HIGHEST] = osPriorityRealtime7,
+    };
+
+    osThreadAttr_t c_tThreadAttribute = {
+        .stack_size = stack_size,
+        .priority = prio_map[prio],
+    };
+
+    *thread = osThreadNew(callback, user_data, &c_tThreadAttribute);
+
+    if(NULL == *thread) {
+        LV_LOG_WARN("Error: Failed to create a cmsis-rtos2 thread.");
+        return LV_RES_INV;
+    }
+
+    return LV_RES_OK;
+
+}
+
+lv_res_t lv_thread_delete(lv_thread_t * thread)
+{
+    osThreadDetach(*thread);
+    osStatus_t status = osThreadTerminate(*thread);
+    if(status == osOK) {
+        return LV_RES_OK;
+    }
+    return LV_RES_INV;
+}
+
+lv_res_t lv_mutex_init(lv_mutex_t * mutex)
+{
+    const osMutexAttr_t Thread_Mutex_attr = {
+        "LVGLMutex",
```

Comment:
Will it be a problem if we create multiple mutexes with the same name?

Reply:
Multiple threads having the same name won't cause any problem at runtime. Even if we want to have unique names, there isn't sufficient input / info in this function.